### PR TITLE
Deletion Marker Fix

### PR DESCRIPTION
**Pull request by mridul-ms**
## What changed?
- Added null checks for optional parameters
- Added backslash end check for registry keys

### CODE REVIEW

File: fixups/RegLegacyFixups/RegistryFixups.cpp [hunk 1]
```diff
@@ -570,7 +570,12 @@ LSTATUS __stdcall RegOpenKeyExFixup(
     Log("[%d] RegOpenKeyEx:\n", RegLocalInstance);
 
 
-    std::string keypath = InterpretKeyPath(key) + "\\" + InterpretStringA(subKey);
+    std::string keypath = InterpretKeyPath(key);
+    if (keypath.size() && keypath.back() != '\\')
```
Comment (ravishroshanms) [inline fixups/RegLegacyFixups/RegistryFixups.cpp:574]: Move this change inside InterpretKeyPath function.

Comment (mridul-ms) [reply fixups/RegLegacyFixups/RegistryFixups.cpp]: Moved.

**Review by ravishroshanms — APPROVED** [review]

File: fixups/RegLegacyFixups/Logging.h [hunk 2]
```diff
@@ -486,6 +486,10 @@ std::string InterpretKeyPath(HKEY key)
     {
         Log("InterpretKeyPath failure.");
     }
+    if (!sret.empty() && sret.back() == '\\')
```
Comment (npuvvada) [inline fixups/RegLegacyFixups/Logging.h:489]: Add it inside try block

**Review by npuvvada — APPROVED** [review]

File: fixups/RegLegacyFixups/Logging.h (hunk 2 above)
Comment (mridul-ms) [reply fixups/RegLegacyFixups/Logging.h]: Done

**Pull request merged**